src/commands/validate_numeric.yml

```
@@ -0,0 +1,16 @@
+description: Validate parameter is a number
+parameters:
+  key:
+    description: The name of the parameter to validate
+    type: string
+  value:
+    description: The value of the parameter to validate
+    type: string
+steps:
+  - run:
+      name: Verify << parameters.key >> is a number
+      command: |
+        if ! echo "<< parameters.value >>" | grep -Eq '^[0-9]+$' ; then
```

```suggestion
        if ! echo "<< parameters.value >>" | grep -Eq '^\d+$' ; then
```

is a possibility, although what you have is fine.